File: ra-front/src/app/model/Robots/ExtraRobotElement.ts
import {ElementFunctionality} from "./ElementFunctionality";

export class ExtraRobotElement{
  id: string;
  name: string;
  dimension: string;
  functionalityList: ElementFunctionality[];

  constructor(name: string = '', dimension: string = '',
              functionalityList: ElementFunctionality[] = []) {
    this.name = name;
    this.dimension = dimension;
    this.functionalityList = functionalityList;
  }
}

File: ra-back/src/main/java/pl/raptors/ra_back/domain/robots/ExtraRobotElement.java
package pl.raptors.ra_back.domain.robots;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@NoArgsConstructor
@Data
@Document(collection = "extra_robot_elements")
public class ExtraRobotElement {

    @Id
    private String id;
    private String name;
    private String dimensions;
    private List<ElementFunctionality> functionalityList;

    public ExtraRobotElement(String name, String dimensions, List<ElementFunctionality> functionalityList) {
        this.name = name;
        this.dimensions = dimensions;
        this.functionalityList = functionalityList;
    }


    public ExtraRobotElement(String name, String dimensions)
    {
        this.name = name;
        this.dimensions = dimensions;
    }
}
